### PR TITLE
Fix to qemu experimental and improvements

### PR DESCRIPTION
**Pull request by wainersm**
This contain a fix to the qemu experimental which currently doesn't get some patches applied.
I took this opportunity to improve a little bit the build scripts for QEMU.

### CODE REVIEW

**Review by GabyCT — APPROVED** [review]

File: tools/packaging/static-build/qemu-virtiofs/Dockerfile [hunk 1]
```diff
@@ -49,23 +49,17 @@ RUN cd  .. && git clone "${QEMU_VIRTIOFS_REPO}" qemu-virtiofs
...
-RUN [ ! -s patches_virtiofs ] || git apply $(cat patches_virtiofs)
-
-RUN cat VERSION | awk 'BEGIN{FS=OFS="."}{print $1 "." $2 ".x"}' > stable_branch
-RUN echo "root/kata_qemu_patches/$(cat stable_branch)/" > patches_qemu_dir
-RUN echo "patches dir $(cat patches_qemu_dir)"
-RUN find "$(cat patches_qemu_dir)" -name '*.patch' -type f |sort -t- -k1,1n > patches_qemu
-RUN echo "Patches to apply for qemu:"
-RUN cat patches_qemu
-RUN [ ! -s patches_qemu ] || git apply $(cat patches_qemu )
+ENV VIRTIOFS_PATCHES_DIR=/root/kata_qemu/patches/${QEMU_VIRTIOFS_TAG}/
```
Comment (bergwolf) [inline tools/packaging/static-build/qemu-virtiofs/Dockerfile:56]: what is `QEMU_VIRTIOFS_TAG` suppose to be set to? It looks like we might end up applying the same patch twice if ` QEMU_VIRTIOFS_TAG` is set to the same as qemu `stable_branch`?

Comment (wainersm) [reply tools/packaging/static-build/qemu-virtiofs/Dockerfile:56]: @bergwolf according to 'versions.yaml` the value of `QEMU_VIRTIOFS_TAG` will be `qemu5.0-virtiofs-with51bits-dax`.

It seems unlikely to have same value of the `stable_branch`. The stable branch will have a value like `5.1.x` and unless it has a branch named `5.1.x` (which is ugly) then the values won't clash. Well, if you want I can put a guard...better safe than sorry :)

Comment (bergwolf) [reply tools/packaging/static-build/qemu-virtiofs/Dockerfile:56]: But we do not have these patches in tree, right? Are these added to apply some external patches? It's a bit confusing though that the version is maintained in tree but no actual patches are committed.

Looks like we should be able to just clone and use `qemu5.0-virtiofs-with51bits-dax` tagged code from `https://gitlab.com/virtio-fs/qemu.git`. Then `VIRTIOFS_PATCHES_DIR` is something obsolete and can be removed?

Comment (wainersm) [reply tools/packaging/static-build/qemu-virtiofs/Dockerfile:56]: This  mechanism to apply virtio-fs specific patches was introduced in commit 04b156f60.

Currently we don't have any specific patch, i.e, a checkout from `qemu5.0-virtiofs-with51bits-dax` branch just works. But suppose that into the future there should be applied patches on `qemu5.0-virtiofs-with51bits-dax` branch which aren't merged into `https://gitlab.com/virtio-fs/qemu.git` first, then we just drop the patches in the `qemu/patches/qemu5.0-virtiofs-with51bits-dax` which they will be applied (no changes to Dockerfile needed).

In this PR I am just keeping the same logic but I can change it. IMO this mechanism can be kept.

Comment (bergwolf) [reply tools/packaging/static-build/qemu-virtiofs/Dockerfile:56]: Fair enough! Thanks for the explanation.

**Review by bergwolf — APPROVED** [review]

**Comment by wainersm** [comment]
@bergwolf thanks for the review and merging. One more question: I believe I need to backport it since it fixes an issue (beyond the improvements). How?

**Pull request merged**